src/librustdoc/html/static/main.js

```diff
@@ -40,6 +40,8 @@
                      "union",
                      "foreigntype"];
 
+    var search_input = document.getElementsByClassName('search-input')[0];
```

Comment:
I think i'm a little fuzzy on JS's scoping/object rules. Does this automatically get updated if someone types in a search, deletes it, then searches for something else? Having everything reference this variable here seems a little off, but my intuition is based around Rust's rules, not JavaScript's. `>_>`

Reply:
Consider it the same as a reference (or a view). You point to an element by you get the accurate information when you access its fields.

Reply:
Makes sense, thanks! C# has a similar system for what they call "reference types".